sdk/src/main/java/io/opentelemetry/sdk/tags/TagMapSdk.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.tags;
+
+import static io.opentelemetry.internal.Utils.checkNotNull;
+
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.tags.Tag;
+import io.opentelemetry.tags.TagKey;
+import io.opentelemetry.tags.TagMap;
+import io.opentelemetry.tags.TagMetadata;
+import io.opentelemetry.tags.TagValue;
+import io.opentelemetry.tags.unsafe.ContextUtils;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+@Immutable
+public class TagMapSdk implements TagMap {
```

Comment:
Why public? So far no need to expose this public.

Reply:
You're right.  I forgot about that.  I changed visibility to package.